Comments/Models/Context.cs

```diff
@@ -113,6 +117,74 @@ public IEnumerable<Location> GetAllCommentsAndQuestionsForDocument(IList<string>
+                    .Select(c=> c.SubmissionComment
+                        .Select(s=> s.Submission)))
+                .IncludeFilter(l => l.Comment.Where(c => c.StatusId == (int)StatusName.SubmittedToLead 
+                                                         && _organisationIDs.Any(o => o.Equals(c.OrganisationId)) 
+                                                         && !_organisationUserIDs.Contains((int)c.OrganisationUserId))
+                    .Select(c=> c.Status))
+                .IncludeFilter(l => l.Comment.Where(c => c.StatusId == (int)StatusName.SubmittedToLead
+                                                         && _organisationIDs.Any(o => o.Equals(c.OrganisationId))
+                                                         && !_organisationUserIDs.Contains((int)c.OrganisationUserId))
+                    .Select(c=> c.OrganisationUser))
+
+                .IncludeFilter(l => l.Question)
+                .IncludeFilter(l => l.Question
+                    .Select(q => q.QuestionType))
+                .IncludeFilter(l => l.Question
+                    .Select(q => q.Answer.Where(a => a.StatusId == (int)StatusName.SubmittedToLead
+                                                     && _organisationIDs.Contains((int)a.OrganisationId) 
+                                                     && !_organisationUserIDs.Contains((int)a.OrganisationUserId))
+                        .OrderByDescending(a=> a.LastModifiedDate).Select(a => a.LastModifiedDate).FirstOrDefault()))
+                .IncludeFilter(l => l.Question
+                    .Select(q => q.Answer.Where(a => a.StatusId == (int)StatusName.SubmittedToLead
+                                                     && _organisationIDs.Contains((int)a.OrganisationId)
+                                                     && !_organisationUserIDs.Contains((int)a.OrganisationUserId))
+                    .Select(a => a.SubmissionAnswer)))
+                .IncludeFilter(l=> l.Question
+                    .Select(q => q.Answer.Where(a => a.StatusId == (int)StatusName.SubmittedToLead
+                                                     && _organisationIDs.Contains((int)a.OrganisationId)
+                                                     && !_organisationUserIDs.Contains((int)a.OrganisationUserId))
+                    .Select(a => a.OrganisationUser)))
+                .OrderBy(l => l.Order)
+                .ToList();
+
+            var filteredLocations = commentsAndAnswers.Where(l =>
+                (l.Order != null) && sourceURIs.Contains(l.SourceURI, StringComparer.OrdinalIgnoreCase));
+
+            var sortedData = filteredLocations.Where(l => l.Comment.Count > 0 || l.Question.Count > 0)
+                .OrderBy(l => l.Order)
+                .ThenByDescending(l =>
+                    l.Comment.OrderByDescending(c => c.LastModifiedDate).Select(c => c.LastModifiedDate)
+                        .FirstOrDefault());
+
+            var convertedData =
```

Comment:
can this conversion to view model move to the service please?

we try and just use the context class purely for database operations, whereas converting to view model and assigning to a new class should live in the service.